### ISSUE TITLE
Switch from awesome_print to amazing_print to avoid deprecation warnings

### ISSUE THREAD

**Issue by qortex** [tilfin/ougai#103]
The `awesome_print` gem is no longer maintained.

A fork has been created that will continue maintaining the gem: [amazing_print](https://github.com/amazing-print/amazing_print).

**The dependency should move to `amazing_print`.**

As of now, `awesome print` gives lots of warning with Ruby 2.7.0 such as (lots of), making ougai practically unusable:
```
/home/mic/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/awesome_print-1.8.0/lib/awesome_print/formatters/base_formatter.rb:113: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
/home/mic/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/awesome_print-1.8.0/lib/awesome_print/inspector.rb:63: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
```

**Comment by qortex** [comment]
See [here](https://github.com/awesome-print/awesome_print/issues/375) for context on the fork.

**Comment by ybiquitous** [comment]
I would appreciate it so much if this proposal would be released, please... 🙏 
(I'm bothered by these too many warnings 😭 )

**Comment by tilfin** [comment]
released it at v1.8.4

**Comment by ybiquitous** [comment]
Thank you so much! 😊
